torch/distributed/_state_dict_utils.py

```diff
@@ -585,6 +596,34 @@ def _broadcast_state_dict(
         _broadcast_tensors(ret, local_state_dict, keys, device, pg)
 
 
+def _distribute_state_dict(
+    full_state_dict: Dict[str, Any],
+    local_state_dict: Dict[str, Any],
+    device: torch.device,
+    pg: Optional[dist.ProcessGroup] = None,
+) -> None:
+    # Distribute tensors in each rank
```

Comment:
Can you add more description for this function?

Reply:
Will add a more detailed description:
Full_state_dict = True but broadcast_from_rank0 = False here. Each rank have full_state_dict. Skip the broadcasting in ``_broadcast_state_dict`` and distribute tensors in each rank